pysat/tests/test_files.py

```diff
@@ -969,6 +974,107 @@ def test_files_when_duplicates_forced(self):
+
+    return 'instrument {}'.format(j)
+
+
+class TestFilesRaceCondition():
+
+    temporary_file_list = False
+
+    def setup(self):
+        """Runs before every method to create a clean testing setup."""
+        # store current pysat directory
+        self.data_path = pysat.data_dir
+        # create temporary directory
+        dir_name = tempfile.gettempdir()
+        pysat.utils.set_data_dir(dir_name, store=False)
+        # create testing directory
+        create_dir(temporary_file_list=self.temporary_file_list)
+
+        # create a test instrument, make sure it is getting files from
+        # filesystem
+        re_load(pysat.instruments.pysat_testing)
+        pysat.instruments.pysat_testing.list_files = list_versioned_files
+        # create a bunch of files by year and doy
+        self.testInst = \
+            pysat.Instrument(inst_module=pysat.instruments.pysat_testing,
+                             clean_level='clean',
+                             temporary_file_list=self.temporary_file_list)
+
+        self.root_fname = ''.join(('pysat_testing_junk_{year:04d}_{month:02d}',
+                                   '_{day:03d}{hour:02d}{minute:02d}',
+                                   '{second:02d}_stuff_{version:02d}_',
+                                   '{revision:03d}.pysat_testing_file'))
+        start = dt.datetime(2007, 12, 30)
+        stop = dt.datetime(2008, 12, 31)
+        create_versioned_files(self.testInst, start, stop, freq='1D',
+                               use_doy=False, root_fname=self.root_fname)
+
+        self.testInst = \
+            pysat.Instrument(inst_module=pysat.instruments.pysat_testing,
+                             clean_level='clean',
+                             update_files=True,
+                             temporary_file_list=self.temporary_file_list)
+
+        print(' '.join(('initial files created in ',
+                        self.testInst.files.data_path)))
+
+    def teardown(self):
+        """Runs after every method to clean up previous testing."""
+        remove_files(self.testInst)
+        del self.testInst
+        re_load(pysat.instruments.pysat_testing)
+        re_load(pysat.instruments)
+        # make sure everything about instrument state is restored
+        # restore original file list, no files
+        pysat.Instrument(inst_module=pysat.instruments.pysat_testing,
+                         clean_level='clean',
+                         update_files=True,
+                         temporary_file_list=self.temporary_file_list)
+        pysat.utils.set_data_dir(self.data_path, store=False)
+
+    def test_race_condition(self):
+        from multiprocessing import Pool
+        processes = 5
+        p = Pool(processes)
+        pysat.file_timeout = 1
+
+        print('beginning tests with file_timeout {}'.format(pysat.file_timeout))
+        result = p.map(create_instrument, range(processes))
+        print(result)
```

Comment:
I think the one last thing that needs to be implemented here is to finish the tests for the race conditions